ccinput/tests/test_psi4.py

```diff
@@ -92,35 +103,285 @@ def test_opt_HF(self):
+        REF = """
+        #File created by ccinput
+
+        memory 1000 mb
+
+        molecule {
+        0 1  
+        O   -0.03589000  -0.05389000   0.00000000
+        H   -0.33841000   0.84272000   0.00000000
+        H    0.90805000   0.01251000   0.00000000
+        }
+
+        frequencies('b3lyp/6-31G')
+        """
+        self.assertTrue(self.is_equivalent(REF, inp.input_file))
+        
+    def test_optfreq_HF(self):
+        params = {
+            "type": "opt+freq",
+            "file": "h2o.xyz",
+            "software": "psi4",
+            "method": "HF",
+            "basis_set": "ccpvdz",
+        }
+        
+        REF="""
+        #File created by ccinput
+
+        memory 1000 mb
+
+        molecule {
+        0 1  
+        O   -0.03589000  -0.05389000   0.00000000
+        H   -0.33841000   0.84272000   0.00000000
+        H    0.90805000   0.01251000   0.00000000
+        }
 
+        optimize('scf/cc-pVDZ')
+        frequencies('scf/cc-pVDZ')
+        """
+        inp = self.generate_calculation(**params)
+        self.assertTrue(self.is_equivalent(REF, inp.input_file))
+    
+    
+    def test_optfreq_DFT(self):
+        params = {
+            "type": "opt+freq",
+            "file": "h2o.xyz",
+            "software": "psi4",
+            "method": "b3lyp",
+            "basis_set": "6-31G",
+        }
+        
+        REF="""
+        #File created by ccinput
+
+        memory 1000 mb
+
+        molecule {
+        0 1  
+        O   -0.03589000  -0.05389000   0.00000000
+        H   -0.33841000   0.84272000   0.00000000
+        H    0.90805000   0.01251000   0.00000000
         }
 
         optimize('b3lyp/6-31G')
+        frequencies('b3lyp/6-31G')
         """
+        inp = self.generate_calculation(**params)
+        self.assertTrue(self.is_equivalent(REF, inp.input_file))
+        
+    def test_optfreq_single_specification(self):
+        params = {
+            "type": "opt+freq",
+            "file": "h2o.xyz",
+            "software": "psi4",
+            "method": "b3lyp",
+            "basis_set": "6-31G",
+            "specifications": "opt(return_wfn=True)",
+        }
+        
+        REF = """
+        #File created by ccinput
 
+        memory 1000 mb
+
+        molecule {
+        0 1  
+        O   -0.03589000  -0.05389000   0.00000000
+        H   -0.33841000   0.84272000   0.00000000
+        H    0.90805000   0.01251000   0.00000000
+        }
+
+        optimize('b3lyp/6-31G', return_wfn=True)
+        frequencies('b3lyp/6-31G')
+        """
+        inp = self.generate_calculation(**params)
         self.assertTrue(self.is_equivalent(REF, inp.input_file))
+      
+    def test_optfreq_multispecs(self):
+        params = {
+            "type": "opt+freq",
+            "file": "h2o.xyz",
+            "software": "psi4",
+            "method": "b3lyp",
+            "basis_set": "6-31G",
+            "specifications": "opt(return_wfn=True) freq(dertype=1 return_wfn=True)",
```

Comment:
What happens if the specifications are `opt(return_wfn=True) dertype=1, return_wfn=True`?

Reply:
It will generate an output with only specifications for the "opt" call. The additional specifications are ignored. 
But, yes thanks for pointing it out!, because in general that is an invalid input. In the scenario where the user would like to add specifications for both calls "opt" and  "freq", then it should be explicitly written as shown in the documentation. I will rise an exception here so that the user knows such input for the specs are wrong. Let me know if you agree or have suggestions.